components/renku_data_services/platform/db.py

```diff
@@ -56,3 +61,112 @@ async def update_config(
+        authz: Authz,
+        project_repo: ProjectRepository,
+    ) -> None:
+        self.session_maker = session_maker
+        self.authz = authz
+        self.project_repo = project_repo
+
+    async def get_redirect_configs(
+        self,
+        user: base_models.APIUser,
+        pagination: PaginationRequest,
+    ) -> tuple[list[models.UrlRedirectConfig], int]:
+        """Get all url redirect configs from the database."""
+        if user.id is None:
+            raise errors.UnauthorizedError(message="You do not have the required permissions for this operation.")
+        if not user.is_admin:
+            raise errors.ForbiddenError(message="You do not have the required permissions for this operation.")
+
+        async with self.session_maker() as session:
+            stmt = select(schemas.UrlRedirectsORM)
+            stmt.limit(pagination.per_page).offset(pagination.offset)
+            stmt_count = select(func.count()).select_from(schemas.UrlRedirectsORM)
+            results = await session.stream_scalars(stmt), await session.scalar(stmt_count)
+            redirects = await results[0].all()
+            return [r.dump() for r in redirects], results[1] or 0
+
+    async def get_redirect_config_by_source_url(
+        self, user: base_models.APIUser, source_url: str
+    ) -> models.UrlRedirectConfig:
+        """Retrieve redirect config for a given source URL."""
+
+        if user.id is None:
+            raise errors.UnauthorizedError(message="You do not have the required permissions for this operation.")
+
+        async with self.session_maker() as session:
+            stmt = select(schemas.UrlRedirectsORM).where(schemas.UrlRedirectsORM.source_url == source_url)
+            result = await session.execute(stmt)
+            url_redirect_orm: schemas.UrlRedirectsORM | None = result.scalars().first()
+
+            if not url_redirect_orm:
+                raise errors.MissingResourceError(
+                    message=f"A redirect for '{source_url}' does not exist or you do not have access to it."
+                )
+
+            # We do not currently check if the user is allowed to access the project, but we could...
+            return url_redirect_orm.dump()
+
+    async def create_redirect_config(
+        self, user: base_models.APIUser, post: models.UnsavedUrlRedirectConfig
+    ) -> models.UrlRedirectConfig:
+        """Create a new URL redirect config."""
+        if user.id is None:
+            raise errors.UnauthorizedError(message="You do not have the required permissions for this operation.")
+        if not user.is_admin:
+            raise errors.ForbiddenError(message="You do not have the required permissions for this operation.")
+
+        async with self.session_maker() as session, session.begin():
+            existing = await session.scalar(
+                select(schemas.UrlRedirectsORM).where(schemas.UrlRedirectsORM.source_url == post.source_url)
+            )
```

Comment:
question: Why not just reuse `get_redirect_config_by_source_url` in this and 1 or 2 other cases in ths same class?